### DISCUSSION TITLE
Add a backup and add a user profile

### DISCUSSION THREAD

**Discussion by kiro5678** [tnodir/fort#150] [Ideas]
I would like to thank the developer for this wonderful program. It outperforms Simplewall, but it lacks some features, such as backing up user settings. The ability to add more than one user profile.

**Comment by ghost** [comment]
Backup settings was already filed by the Dev in https://github.com/tnodir/fort/issues/19 so it is a duplicate.
But Fort Firewall stores the files (SQLite Databases) in `C:\ProgramData\Fort Firewall`:
- FortFirewall.config = Programs, AppGroup, IPs, Zones
- FortFirewall.ini =  Settings from Options tab and Graph settings.
- FortFirewall.stat =  Statistics
- FortFirewall.stat-block = Connections Blocked
- Cache = Programs identified by Fort, Descriptions, icons and all that for Programs window.

and the other user settings for hotkeys and window positions and all that are in `%localappdata%\Fort Firewall`.

So, they are easy to manually backup.

About Profiles, what firewall offers profiles? Because [AppGroups](https://github.com/tnodir/fort/wiki/User-Guide#application-groups) can achieve the same and probably be better than any other Firewall, since Fort has wildcard support and child processes and svchost services and all that.
But you can easily add all the programs you want with wildcards to an AppGroup, and use the AppGroups Enable state to affect invidual Programs and Wildcard programs if necessary, after that, you use the tray icon or keyboard shortcuts or even by schedule to easily switch between AppGroups like you would with a 'Profile', turn off a group, turn on the other.

**Comment by tnodir** [comment]
@kiro5678 Can you please describe how do you use profiles in Simplewall?

Where are the profiles in Simplewall's UI?

**Comment by kiro5678** [comment]
> @kiro5678 Can you please describe how do you use profiles in Simplewall?

There is no user profile in Simplewall

**Comment by kiro5678** [comment]
@tnodir The user can switch to any of the user profiles, as the user profile maintains the user configurations.  For example, User Profile 1 holds games blocked while User Profile 2 contains user settings to allow games.  This is a method similar to chrome. I hope you understand me well

**Comment by tnodir** [comment]
Backup is #19.
Sort of profiles are planned in #2.

**Comment by tnodir** [comment]
> For example, User Profile 1 holds games blocked while User Profile 2 contains user settings to allow games. This is a method similar to chrome.

Hm I don't know, I don't use Chrome.

Did you try the App Groups in Fort?

**Comment by kiro5678** [comment]
Yes, I tried that.  The way an application groups works is different from a user profile.  The user profile keeps all user configurations like application groups, allowed and blocked applications, blocked IP addresses and all user settings.  Where the user can create more than one user profile.  Please excuse me if there are misspelled words, I am using Google Translate

**Comment by tnodir** [comment]
> The user profile keeps all user configurations like application groups, allowed and blocked applications, blocked IP addresses and all user settings.

Sorry, this is out of plan and requires some redesign.

You can use `--profile <PATH>` command line argument, but it should be used by Service too. So, not easy for ordinary user.

Also you can use `FortFirewall.exe.example.ini` in the Fort's installed directory to create `FortFirewall.exe.ini` and specify `profileDir` and `userDir` directories.

**Reply by kiro5678** [reply]
I thought about that. This feature is an added luxury for the user, but it is also a crazy idea, so don't bother with it.
